src/effects/native/flangereffect.cpp

```diff
@@ -84,38 +71,49 @@ void FlangerEffect::processChannel(const ChannelHandle& handle,
     Q_UNUSED(handle);
     Q_UNUSED(enableState);
     Q_UNUSED(groupFeatures);
-    Q_UNUSED(sampleRate);
-    CSAMPLE lfoPeriod = m_pPeriodParameter->value();
-    CSAMPLE lfoDepth = m_pDepthParameter->value();
-    // Unused in EngineFlanger
-    // CSAMPLE lfoDelay = m_pDelayParameter ?
-    //         m_pDelayParameter->value().toDouble() : 0.0f;
 
-    // TODO(rryan) check ranges
-    // period needs to be >=0
-    // delay needs to be >=0
-    // depth is ???
+    // TODO: remove assumption of stereo signal
+    const int kChannels = 2;
+
+    const unsigned int lfoAmplitude =
+        kLfoAmplitude / sampleRate / kChannels;
+    const unsigned int averageDelayLength =
+        kAverageDelayLength / sampleRate / kChannels;
+    const unsigned int maxDelay =
+        kMaxDelay / sampleRate / kChannels;
+
+    // The parameter minimum is zero so the exact center of the knob is 2 beats.
+    CSAMPLE beatsPerPeriod = std::max(m_pPeriodParameter->value(), 0.05);
+    CSAMPLE lfoPeriod;
+    if (groupFeatures.has_beat_length) {
+        lfoPeriod = beatsPerPeriod * groupFeatures.beat_length;
+    } else {
+        // Act as if the input is 100 BPM, which is a totally arbitrary tempo.
+        lfoPeriod = beatsPerPeriod
+            / 100.0 * 60.0 * sampleRate / kChannels;
+    }
+
+    CSAMPLE lfoDepth = m_pDepthParameter->value();
 
     CSAMPLE* delayLeft = pState->delayLeft;
     CSAMPLE* delayRight = pState->delayRight;
 
-    const int kChannels = 2;
     for (unsigned int i = 0; i < numSamples; i += kChannels) {
         delayLeft[pState->delayPos] = pInput[i];
         delayRight[pState->delayPos] = pInput[i+1];
 
-        pState->delayPos = (pState->delayPos + 1) % kMaxDelay;
+        pState->delayPos = (pState->delayPos + 1) % maxDelay;
 
         pState->time++;
         if (pState->time > lfoPeriod) {
             pState->time = 0;
         }
 
         CSAMPLE periodFraction = CSAMPLE(pState->time) / lfoPeriod;
-        CSAMPLE delay = kAverageDelayLength + kLfoAmplitude * sin(M_PI * 2.0f * periodFraction);
+        CSAMPLE delay = averageDelayLength + lfoAmplitude * sin(M_PI * 2.0f * periodFraction);
```

Comment:
@JosepMaJAZ I tried turning the center value (averageDelayLength) and amplitude into effect parameters but it didn't seem very useful to me. Maybe I just didn't choose appropriate scales for them. Do you have suggestions?